Backports for 1.12.0-beta3

KristofferC opened the pull request:
Backported PRs:
- [x] #57523 <!-- Remove usages of weak symbols -->
- [x] #58166 <!-- staticdata: fix many mistakes in staticdata stripping -->
- [x] #58200 <!-- Narrow `@inbounds` annotations in `accumulate.jl` to only indexing calls -->
- [x] #58057 <!-- fix allocations macro with more functions -->
- [x] #58212 <!-- improve type stability of `stat` -->
- [x] #58216 <!-- optimize Type method table queries and insertions -->
- [x] #58209 <!-- Specialize `one` for the `SizedArray` test helper -->
- [x] #58196 <!-- Subtype: enable more Tuple related fast path. -->
- [x] #58218 <!-- [REPLCompletions] use better version of listing module imports for completions -->
- [x] #58248 <!-- Update MozillaCACerts_jll to 2025-02-25 -->
- [x] #58247 <!-- Update deps to OpenSSL 3.5.0 -->
- [x] #58260 <!-- add PARTITION_KIND_BACKDATED_CONST to UndefVarError_hint -->
- [x] #57129 <!-- clarify that time_ns is monotonic -->
- [x] #58113 <!-- doc: cross-reference `bind` in `Channel` method doc string -->
- [x] #58134 <!-- Note annotated string API is experimental in Julia 1.11 in HISTORY.md -->
- [x] #58219 <!-- Tracy: annotate timings in `JIT_Total` zone with function names -->
- [x] #58265 <!-- OncePerX: Improve initializer type for DataType constructors -->
- [x] #58271 <!-- bpart: Fix a hang in a particular corner case -->
- [x] #58254 <!-- Strengthen language around `@assume_effects` :consistent -->
- [x] #58210 <!-- Print fallback REPL prompt before user input -->
- [x] #58121 <!-- fix: give access to displaysize when displaying at REPL -->
- [x] #58273 <!-- Prevent type infer hang of "simple" recursive functions -->
- [x] #58307 <!-- Fix lowering failure with type parameter in opaque closure -->
- [x] #58258 <!-- Make build_id.lo more random -->
- [x] #58266 <!-- add backdate_admonition to depwarn=error -->
- [x] #58325 
- [x] #58327 
- [x] #58314

Need manual backport:
- [ ] #57410 <!-- codegen: cleanup gcstack call frames somewhat earlier -->
- [ ] #58157 <!-- add showing a string to REPL precompile workload -->
- [ ] #58205 <!-- reduce places where Builtins are listed -->
- [ ] #58197 <!-- trimming: Remove `init.c` from test -->

Contains multiple commits, manual intervention needed:
- [ ] #57542 <!-- staticdata: Refactor sysimage loading -->
- [ ] #58141 <!-- Make trimmed binaries initialize on first call -->
- [ ] #58231 <!-- Refactor julia initialization by finding and loading the sysimage earlier -->
- [x] #58014 <!-- trimming: Support finalizers -->
- [x] #57763 <!-- add the ability to specify the external name of a ccallable -->

Non-merged PRs with backport label:
- [x] #58311 <!-- trimming: follow-ups to finalizer support -->
- [ ] #58308 <!-- only warn about Vararg wrapping once -->
- [ ] #58301 <!-- Detect Apple M4 and some related changes -->
- [ ] #58261 <!-- fix isconst definition/accessor issues with binding partitions -->
- [ ] #58110 <!-- relax dispatch for the `IteratorSize` method for `Generator` -->
- [ ] #58108 <!-- Base.get_extension & Dates.format made public -->
- [ ] #58070 <!-- fix called-argument analysis for calls with splat -->
- [ ] #58038 <!-- strings/cstring: `transcode`: prevent Windows sysimage invalidation -->
- [ ] #57886 <!-- only show backdated binding warning when deprecation warnings are enabled -->
- [ ] #57604 <!-- `@nospecialize` for `string_index_err` -->
- [ ] #57490 <!-- export `maxthreadid` from `Threads` -->
- [ ] #57481 <!-- juliac: Add rudimentary Windows support -->
- [ ] #57454 <!-- If the user explicitly asked for 1 thread don't add an interactive one. -->
- [ ] #57422 <!-- Fix performance regression in hvcat of simple matrices -->
- [ ] #57366 <!-- Use ptrdiff_t sized offsets for gvars_offsets to allow large sysimages -->
- [ ] #57143 <!-- Add JuliaLang/JuliaSyntax.jl#525 to NEWS.md, flisp parser, and REPL -->

gbaraldi committed:
Add hook to initialize Julia on-the-fly during thread adoption (#56334)

This adds `jl_pgcstack_default_func` to allow an uninitialized runtime to make it into `jl_autoinit_and_adopt_thread`

The init changes themselves are left to #57498

---------

Co-authored-by: Cody Tapscott <topolarity@tapscott.me>

topolarity committed:
Remove usages of weak symbols (#57523)

Inspired by a bug I hit on MinGW recently with weak symbol resolution.
I'm not sure why we started using these in
70000ac7c3d5d5f21e42555cdf99e699a246f8ec, since I believe we should be
able to lookup these symbols just fine in our `jl_exe_handle` as long as
it was linked / compiled with `-rdynamic`

Migrating away from weak symbols seems a good idea, given their uneven
implementation across platforms.

(cherry picked from commit 4a6ada6eb36eaeb9bde9b94c01bcf6f0e3ce7ae6)

JeffBezanson committed:
add the ability to specify the external name of a ccallable

staticfloat committed:
test/trimming: Use `SHLIB_EXT` in Makefile

(cherry picked from commit 3204987f8a19f9833678d7936a02e53d904c4d4b)

JeffBezanson committed:
remove unused argument from jl_generate_ccallable

topolarity committed:
trim: Add `Core.finalizer` support

This adds the changes required to guess the MethodInstance that the
runtime will eventually invoke for this call, enqueue it for codegen,
and then verify its presence in the verifier.

JeffBezanson committed:
some fixes

topolarity committed:
typeinfer: add separate compile workqueue for native / call_latest code

topolarity committed:
Make `empty!(::BitSet)` work during bootstrap

topolarity committed:
Collect invoke only for `ftyp === typeof(finalizer)`

Otherwise this will match `ftyp === Builtin`, which is pretty silly
since that could just as well be any other built-in.

topolarity committed:
typeinfer: Work around inference bug

See https://github.com/JuliaLang/julia/issues/58143 for details.

topolarity committed:
Adjust `--trim` tests for new functionality

gbaraldi committed:
Make trimmed binaries initialize on first call (#58141)

vtjnash committed:
staticdata: fix many mistakes in staticdata stripping (#58166)

While #58156 was supposed to be fairly trivial, it ended up uncovering
many bugs in the unusual ways that trimming and binding partition had
been added to the serialization format. This attempts to reorganize them
to be handled more consistently and with fewer mistakes.

(cherry picked from commit 828e3a1a7b56ab746b721ba110b4b7bd17cb815b)

adienes committed:
Narrow `@inbounds` annotations in `accumulate.jl` to only indexing calls (#58200)

`op` should not be `inbounds`-ed here

the test case I added might technically be an illegal (or at least bad)
use of `@propagate_inbounds` in case anyone has a suggestion for a more
natural test case, but nonetheless it demonstrates getting a
`BoundsError` instead of a segfault (worse) or incorrect / junk data
(more worse), both of which happen on master

(cherry picked from commit bdab032dbd80b6469353cd7fb8389c09512b2a9f)

vtjnash committed:
"improve" allocations macro with more functions (#58057)

Change the implementation of the allocations to always add a function
call (closure) wrapper so that the inner code can be optimized, even if
the outer scope is `@latestworld`. The implementation creates a closure
if necessary (the code is complex) or just makes a call if the
expression is simple (just a call on symbols). Many packages in the
wild are observed to already be doing this themselves.

(cherry picked from commit a47b58f2668436fa78563d08ec5157d523eae392)

mlechu committed:
Tracy: annotate timings in `JIT_Total` zone with function names

Co-authored-by: Cody Tapscott <topolarity@tapscott.me>
(cherry picked from commit 485575c0f0a8432882da69dd0cac20159232bbb8)

aviatesk committed:
improve type stability of `stat` (#58212)

Make sure that its return type is inferred to be `String`. Allows JET to
not report false positive errors from `@report_call
Downloads.download(::String)`

(cherry picked from commit ead23dbe1032d96476722a3d6d62c38a7f17f18e)

vtjnash committed:
optimize Type method table queries and insertions (#58216)

Ensure a total split of constructors and non-constructors, so they do
not end up seaching the opposing table. Instead cache all kind lookups
as keyed by typename(Type). Not really needed on its own, but it avoids
a minor performance regression in
https://github.com/JuliaLang/julia/pull/58131.

(cherry picked from commit d9925544e0ca3023f9f35196ab2b4694dfa8dd04)

jishnub committed:
Specialize `one` for the `SizedArray` test helper (#58209)

Since the size of the array is encoded in the type, we may define `one`
on the type. This is useful in certain linear algebra contexts.

(cherry picked from commit d9fafab2e0c5e15eed0c20082755a1c685d4f5b3)

N5N3 committed:
Subtype: enable more Tuple related fast path. (#58196)

Fix the subtyping hang found in
https://github.com/JuliaLang/julia/issues/58115#issuecomment-2821388658

(cherry picked from commit c9ad04dcc73256bc24eb079f9f6506299b64b8ec)

vtjnash committed:
[REPLCompletions] use better version of listing module imports for completions (#58218)

The test for this crashes on my machine, leading me to notice that the
implementation of this function was nonsense. Fortunately, we already
had a more correct implementation of this function in the REPL docview
code that we could borrow from to fix this.

While here, also filter out macros, since those are rather strange
looking to see appearing in the results. We could alternatively use
`Base.is_valid_identifier`, since the main point here is to print
functions that are accessible in the module by identifier.

```
julia> @eval Base.MainInclude export broken

julia> broken = 2
2

julia>  ?(┌ Error: Error in the keymap
│   exception =
│    UndefVarError: `broken` not defined in `Base.MainInclude`
```

(cherry picked from commit 8780aa93ac84198bc7ebcbdd784f26041c99dea5)

eschnett committed:
Update MozillaCACerts_jll to 2025-02-25 (#58248)

(cherry picked from commit 4616aa270c81427413765d065db960875b6afd59)

eschnett committed:
Update deps to OpenSSL 3.5.0 (#58247)

(cherry picked from commit dc8bc7366c5bb6bac764f7b18a581640a141b2c3)

vtjnash committed:
add PARTITION_KIND_BACKDATED_CONST to UndefVarError_hint (#58260)

Wording updated based upon timholy's suggestion in #57969.

(cherry picked from commit 1aeea1979fac8f8cc9ffcc89d4c62830be55bebf)

stevengj committed:
clarify that time_ns is monotonic (#57129)

(cherry picked from commit b9a0497e96286fa3e43740294e2c5f6b17b70ae2)

nsajko committed:
doc: cross-reference `bind` in `Channel` method doc string (#58113)

(cherry picked from commit d4f2e8ab3758bc614c80535b13d487cac55a6c8b)

nhz2 committed:
Note annotated string API is experimental in Julia 1.11 in HISTORY.md (#58134)

(cherry picked from commit d46b665067bd9fc352c89c9d0abb591eaa4f7695)

topolarity committed:
OncePerX: Improve initializer type for DataType constructors (#58265)

Generalized from https://github.com/JuliaData/Parsers.jl/pull/197/

(cherry picked from commit 1f1839129b6efbf6f280b56d71f761642763a445)

Keno committed:
bpart: Fix a hang in a particular corner case (#58271)

The `jl_get_binding_partition_with_hint` version of the bpart lookup did
not properly set the `max_world` of the gap, leaving it at `~(size_t)0`.
Having a `gap` with that `max_world`, but with `gap.replace` set is
inconsistent and ended up causing an integer overflow downstream in the
computation of `expected_prev_min_world`, which looked like a concurrent
modification causing an infinite retry. Fixes #58257

(cherry picked from commit fc8e6e731c59f6b62f695a6616f37bfb951d7dca)

Keno committed:
Strengthen language around `@assume_effects` :consistent (#58254)

The current language of `:consistent` for assume_effects is problematic,
because it explicitly promises something false: That you're allowed to
mark a method as `:consistent` even in the face of redefinitions.

To understand this problem a bit better, we need to think about the
primary use case of the `:consistent` annotation: To perform constant
propagation evaluation of fuctions with constant arguments. However,
since constant evaluation only runs in a single world (unlike the
abstract interpreter, which symbolically runs in multiple worlds and
keeps track of which world range its result is valid for), we run into a
bit of a problem. In principle, for full correctness and under the
current definition of consistentcy, we can only claim that we know the
result for a single world age (the one that we ran in). This is
problematic, because it would force us to re-infer the function for
every new world age.

In practice however, ignoring `@assume_effects` for the moment, we have
a stronger guarantee. When inference sets the :consistent effect bit, it
guarantees consistentcy across the entire computed min_world:max_world
range. If there is a redefinition, the world range will be terminated,
even if it is `:consistent` both before and after and even if the
redefinition would not otherwise affect inference's computed
information. This is useful, because it lets us conclude that the
information derived from concrete evluation is valid for the entire
min_world:max_world range of the corresponding code instance.

Coming back to `@assume_effects` we run into a problem however, because
we have no way to provide the required edges to inference. In practice
inference will probably be able to figure it out, but this is
insufficient as a semantic guarantee. After some discussion within the
compiler team, we came to the conclusion that the best resolution was to
change the documented semantics of `@assume_effects :consistent` to
require consistent-cy across the entire defined world range of the
method, not just world-age-by-world-age. This is a significantly
stronger guarantee, but appears necessary for semantic correctness. In
the future we may want to consider `:consistent` annotations for
particular if blocks, which would not require the same restrictions
(because it would still rely on inference to add appropriate edges for
redefinitions).

Closes #46156

(cherry picked from commit d7cd2710f6d919cd99e3b26a8056bbaeee74560c)

Keno committed:
Add manual chapter on world age and binding partition (#58253)

Co-authored-by: Alex Arslan <ararslan@comcast.net>
Co-authored-by: Nick Robinson <npr251@gmail.com>

xal-0 committed:
Print fallback REPL prompt before user input (#58210)

Using the fallback REPL interactively prints the prompt only after user
input, making the scrollback confusing:
```
$ JULIA_FALLBACK_REPL=1 julia
1 + 2
julia> 3
```

---------

Co-authored-by: Jameson Nash <vtjnash@gmail.com>
Co-authored-by: Dilum Aluthge <dilum@aluthge.com>
(cherry picked from commit ccef01a75367acede2f7325416e54407a8255408)

topolarity committed:
Refactor julia initialization by finding and loading the sysimage earlier (#58231)

This changes the order of initialization slightly. The goal is to be
able to initialize executables and share libraries without having to
explicitly call init functions.

This still doesn't address the thread safety

topolarity committed:
trimming: Remove `init.c` from test (#58197)

This accidentally diverged when #58141 was merged without adjusting the
test to match (these tests shouldn't be re-implementing the `juliac`
link step anyway, but they currently do)

Unfortunately, this hid a bug where `jl_pathname_for_handle` does not
understand the main executable.

rfourquet committed:
fix: give access to displaysize when displaying at REPL (#58121)

In #53959, an new `LimitIO` object was defined to display at the REPL,
but it hid the displaysize of the underlying `io`. This patch just
forwards this information.

Before:
```
julia> Dict(1 => fill(UInt(0), 4))
Dict{Int64, Vector{UInt64}} with 1 entry:
  1 => [0x0000000000000000, 0x0000000000000000, 0x0000000000000000, 0x000000000…
```
After:
```
Dict{Int64, Vector{UInt64}} with 1 entry:
  1 => [0x0000000000000000, 0x0000000000000000, 0x0000000000000000, 0x0000000000000000]
```

(cherry picked from commit 35bfba9229ef0a07799ef1e493937f97829d5242)

vtjnash committed:
Prevent type infer hang of "simple" recursive functions (#58273)

Prevent infinite inference when dealing with LimitedAccuracy by
declining to cache them. Presence in the cache triggers further
compilation to resolve it, so removing infinite work from the cache also
prevents infinite work at pre-compile/jit compile time.

Fix #57098
Fix #57873

(cherry picked from commit 48bd67353ddf660469e8b3d5b33eff8b6e36bcf2)

topolarity committed:
trimming: follow-ups to finalizer support

oscardssmith committed:
improve isdefined precision for 0 field types (#58220)

alternate to https://github.com/JuliaLang/julia/pull/58214.

---------

Co-authored-by: Jeff Bezanson <jeff.bezanson@gmail.com>

mlechu committed:
Fix lowering failure with type parameter in opaque closure (#58307)

(cherry picked from commit 7b64cec5385d9099762ad7449c340eaac4fccb41)

gbaraldi committed:
Make build_id.lo more random (#58258)

This does not fix the underlying issue that can occur here which is a
collision of build_ids.lo between modules in IR decompression. Fixing
that requires a somewhat significant overhaul to the serialization of IR
(probably using the module identity as a key). This does mean we use a
lot more of the bits available here so it makes collisions a lot less
likely( they were already extremely rare) but hrtime does tend to only
use the lower bits of a 64 bit integer and this will hopefully add some
more randomness and make this even less likely

(cherry picked from commit 71574073759506b0a3171640ea5e1b468f2757c9)

vtjnash committed:
add backdate_admonition to depwarn=error (#58266)

This is close to the expected behavior after deprecations are removed
(other than that the b->globalref->mod in the printed message here will
be the source module instead of the destination module, which may
sometimes cause confusing printing here, but probably rarely). I also
needed this recently to find a place this warning occurred, so I think
it should be merged now and get feedback later.

Closes #57969

(cherry picked from commit 06821587b0fde7f87432e90d50158c562ac61c0d)

aviatesk committed:
docs: some minor follow-up to JuliaLang/julia#58253 (#58314)

- indent the numbered list so that it is rendered nicely
- add the world age docs to the manual index

Keno committed:
Add 1.12 NEWS entry on binding partition (#58297)

Will need to be manually backported since this is in NEWS.md on 1.12
rather than HISTORY. Closes #57596

---------

Co-authored-by: Shuhei Kadowaki <40514306+aviatesk@users.noreply.github.com>

KristofferC commented:
@nanosoldier `runtests(["SIMDTypes", "LazyBroadcast", "DontMaterialize", "SyntaxTree", "MistyClosures", "Fuzzy", "ModuleDocstrings", "Syslogs", "TypeTree", "CatViews", "TraceFuns", "Bits", "WidthLimitedIO", "FindDefinition", "MultiThreadedCaches", "ExprTools", "FuzzyCompletions", "CoverageTools", "SingleFloats", "WithAlloc", "StringAlgorithms", "CountFlops", "InterProcessCommunication", "OverflowContexts", "CellLists", "Adapt", "MallocArrays", "MappedArrays", "PtrArrays", "InternedStrings", "CompTime", "EnzymeCore", "ReferenceImplementations", "Recyclers", "Infinities", "RNGPool", "OperatorScaling", "ThreadLocalCounters", "BracedErrors", "ChangePrecision", "OptimalSortingNetworks", "TypeStability", "FunctionFusion", "RoundingEmulator", "NarrativeTest", "RuntimeGeneratedFunctions", "StarAlgebras", "KeywordCalls", "YaoHIR", "Match", "Spec", "FieldFlags", "Abaco", "Baobzi", "TestEnv", "Umlaut", "BSON", "Interfaces", "CommonRLInterface", "LMDB", "JACC", "GAFramework", "LoweredCodeUtils", "NormalHermiteSplines", "FileCmp", "LambertW", "TypedSyntax", "PlutoMonacoEditor", "CellularAutomata", "ScikitSpatial", "MixedStructTypes", "StaticArraysBlasInterfaces", "RBNF", "Cliffords", "Tracy", "TrixiBase", "OpenQASM", "SimJulia", "Colors", "DRIPs", "NL2sol", "BitBasis", "GridMaps", "CircularArrayBuffers", "MaybeInplace", "SimpleExpressions", "Mueller", "BibInternal", "JuliaWorkspaces", "SimpleI18n", "GeometryTypes", "FixedPointDecimals", "StaticLint", "AStarGridSearch", "QuanticsGrids", "ConstraintCommons", "SHTns", "Divergences", "BioAtomsCount", "RollingWindowArrays", "BioVossEncoder", "MCP2221Driver", "GitCommand", "BioMarkovChains", "HypertextTemplates", "GraphQLGen", "ADTypes", "DiffieHellman", "PolarizedTypes", "DimensionfulAngles", "Binning2D", "Accessors", "XDiag", "FCSFiles", "ODEInterface", "RustFFT", "DocSeeker", "CayleyMengerDeterminant", "DispatchDoctor", "SigmaProofs", "Cosmology", "BoundedDegreeGraphs", "NiLang", "DynamicQuantumCircuits", "StructArrays", "RNGTest", "SpectralKit", "SMCExamples", "SquashFS", "ShuffleProofs", "InPartS", "FiniteDifferences", "GenericSchur", "PlayingCards", "GenericLinearAlgebra", "NearestNeighborDescent", "InteratomicPotentials", "AllocCheck", "MultiBroadcastFusion", "Stencils", "TerminalGat", "MarchingCubes", "UCX", "ADOLC", "OpenSSLGroups", "Polyester", "LatinHypercubeSampling", "CallableExpressions", "UnitfulAssets", "NestedGraphs", "OpticalPropagation", "InteractiveErrors", "DifferentiableFlatten", "GridWorlds", "WannierIO", "DuckDB", "AtomicSymmetries", "MolecularMinimumDistances", "OpenDSSDirect", "Proj", "DictArrays", "NeidArchive", "FastCholesky", "QuasiArrays", "QEDcore", "LinkedInAPI", "TypedMatrices", "QXZoo", "AccessorsExtra", "NLSolvers", "PkgJogger", "ConvexHulls2d", "AdmittanceModels", "Toolips", "Presentation", "QEDprocesses", "Term", "YAAD", "LinearElasticityBase", "QEDfields", "CometLogger", "BurrowsWheelerAligner", "DoubleFloats", "LinearAlgebraForCAP", "EcRequests", "UnitfulLinearAlgebra", "Norg", "MathematicalSystems", "SparseIR", "EnergyExpressions", "CCBlade", "PyCallJLD2", "XmlStructWriter", "AtiyahBott", "ScHoLP", "FiberNlse", "LandauDistribution", "ToyPublicKeys", "Kronecker", "BibParser", "AngularMomentumAlgebra", "DataToolkit", "Bibliography", "AbstractNeuralNetworks", "MvNormalCalibration", "NomnomlJS", "QuantumAlgebra", "SparseMatricesCOO", "Quante", "IntervalLinearAlgebra", "SymPyCore", "RationalPolygons", "TrustRegionMethods", "Rotations", "ParaReal", "MonotonicSplines", "Air", "Ephemerides", "HopTB", "Tensors", "Experimenter", "NCDatasets", "CCDReduction", "SymFEL", "FrechetDist", "FinancialToolbox", "Seaborn", "TensorCrossInterpolation", "JuDGE", "NeutralLandscapes", "BlackBoxOptim", "SLEEFMath", "QEDevents", "ConstrainedControl", "SymPyPythonCall", "QuantumElectrodynamics", "MatrixFactorizations", "ConstrainedDynamics", "PlutoTeachingTools", "NeptuneAILogger", "SymbolicWedderburn", "SoleBase", "SimilaritySearch", "DataToolkitCommon", "ConstraintDomains", "MonteCarloSummary", "TemporalGPs", "AbstractLogic", "ShiftedProximalOperators", "SparseMatrixColorings", "SunAsAStar", "StaticWebPages", "MultiScaleTreeGraph", "Ipaper", "PRASCapacityCredits", "SpatialEcology", "ReinforcementLearningTrajectories", "AutomotiveSimulator", "StrategicGames", "AbsSmoothFrankWolfe", "ImplicitGlobalGrid", "FlashAttentionWrapper", "SlidingDistancesBase", "BitSAD", "LaserTypes", "RangeEnclosures", "UMAP", "NeumannKelvin", "ConstitutiveModels", "SymbolicIndexingInterface", "InventoryManagement", "Octavian", "PyPlotUtils", "GEMPIC", "VlasiatorPyPlot", "BayesianQuadrature", "Santiago", "ZarrDatasets", "DocstringAsImage", "MatrixProfile", "DifferentiableFrankWolfe", "PowerModelsAnnex", "PlutoPlotly", "PolynomialAmoebas", "GEOTRACES", "Mango", "EnergyModelsRenewableProducers", "WaterModels", "MTH229", "Gaugefields", "SignalTemporalLogic", "DACE", "Maxnet", "MarkovChainHammer", "InfiniteOpt", "ExplainableAI", "Galley", "Mice", "FastTransforms", "TransmuteDims", "TensorCast", "AbstractCosmologicalEmulators", "FeynmanDiagram", "ProfileView", "TensorKitSectors", "Sensemakr", "KiteUtils", "ArDCA", "EtherSPH", "RegNets", "SimulationLogs", "Jadex", "InfrastructureSystems", "EmpiricalPotentials", "ElementaryFluxModes", "ApproxFunFourier", "XCALibre", "FinEtools", "FiniteElementContainers", "CSDP", "KomaMRICore", "OSQP", "SimSpread", "HuggingFaceDatasets", "ReferenceFiniteElements", "TensorOperationsTBLIS", "PoincareInvariants", "LowLevelParticleFilters", "EwaldSummations", "SymbolicUtils", "MRFingerprintingRecon", "TransitionsInTimeseries", "PsychometricsBazaarBase", "PassiveTracerFlows", "DecisionMakingPolicies", "SparseKmeansFeatureRanking", "GraphsOptim", "LinearRegressionKit", "CaratheodoryFejerApprox", "RecurrentLayers", "SUNRepresentations", "ODEInterfaceDiffEq", "ODE", "ImageSegmentation", "ConstraintExplorer", "MPIMeasurements", "Gadfly", "ParallelAnalysis", "DifferentiableMetabolism", "ACTRSimulators", "DynACof", "MCMCDebugging", "JOLI", "PALEOboxes", "Jabalizer", "Polynomials4ML", "OrdinaryDiffEqExtrapolation", "CategoryData", "Wflow", "PowerSystemCaseBuilder", "WordCloud", "Qaintmodels", "IterativeLQR", "PyBraket", "TaylorInversion", "UnROOT", "FusibleBroadcasts", "TimeSeriesClassification", "GMT", "NDTensors", "JUDI", "ConstrainedDynamicsVis", "IESopt", "Spehulak", "RHEOS", "ProbabilisticCircuits", "LatticeModels", "EvoDynamics", "RingStarProblems", "Tasmanian", "PALEOsediment", "ComputerAdaptiveTesting", "ElectronLiquid", "DynamicMovementPrimitives", "QSFit", "EnergySamplers", "DeconvOptim", "FastMPOContractions", "GeometricalOptics", "StartUpDG", "MarginalLogDensities", "PowerPlots", "BoundaryValueDiffEqMIRKN", "ReinforcementLearningFarm", "ExpressionTreeForge", "GeoStatsFunctions", "GeoEstimation", "GridapTopOpt", "GtkUtilities", "SwitchOnSafety", "PALEOmodel", "FluxTraining", "PartiallySeparableNLPModels", "LogicCircuits", "FastBEAST", "KernelInterpolation", "WaveOpticsPropagation", "MGVI", "DistributedStwdLDA", "Swalbe", "ReinforcementLearningCore", "SymbolicAnalysis", "GenericCharacterTables", "MultiStateSystems", "ReinforcementLearningZoo", "TropicalNN", "SpinGlassNetworks", "NonconvexJuniper", "PortfolioAnalytics", "SolverBenchmark", "HetaSimulator", "MNPDynamics", "vSmartMOM", "GNNGraphs", "RvLineList", "DiffusionGarnet", "FractionalSystems", "Bactos", "WGPUgfx", "GeneticsMakie", "OpenQuantumSystems", "FSimPlots", "CellMLToolkit", "NuclearToolkit", "ChargeTransport", "Fable", "Microstructure", "SwissVAMyKnife", "CDGRNs", "Biofilm", "DiskArrayEngine", "RigorousInvariantMeasures", "MAGEMinApp", "CropRootBox", "CalibrateEmulateSample", "MRINavigator", "SBMLToolkitTestSuite"], vs = ":release-1.11")`

KristofferC committed:
bump Pkg to latest v1.12

nanosoldier commented:
The package evaluation job [you requested](https://github.com/JuliaLang/julia/pull/58270#issuecomment-2850152479) has completed - possible new issues were detected.
The [**full report**](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/9e913d7_vs_2d89891/report.html) is available.

<details><summary>Report summary</summary>

## ❗ Packages that crashed

**14 packages crashed only on the current version.**

 - The process was aborted: 1 packages
 - Invalid LLVM IR was generated: 1 packages
 - An internal error was encountered: 9 packages
 - GC corruption was detected: 2 packages
 - A segmentation fault happened: 1 packages


## ✖ Packages that failed

**362 packages failed only on the current version.**

 - Package has syntax issues: 5 packages
 - Package fails to precompile: 91 packages
 - Illegal method overwrites during precompilation: 12 packages
 - Package has test failures: 65 packages
 - Package tests unexpectedly errored: 137 packages
 - Package is using an unknown package: 1 packages
 - Networking-related issues were detected: 1 packages
 - There were unidentified errors: 2 packages
 - Tests became inactive: 3 packages
 - Test duration exceeded the time limit: 41 packages
 - Test log exceeded the size limit: 4 packages

10 packages failed on the previous version too.

## ✔ Packages that passed tests

60 packages passed tests on the previous version too.

## ➖ Packages that were skipped altogether

1 packages were skipped on the previous version too.



</details>

gbaraldi committed:
staticdata: Refactor sysimage loading (#57542)

Introduce `jl_image_buf_t` to represent the in-memory details of an
unparsed system image and clean-up several global variables and split
loading logic in staticdata.c so that we get (almost) everything we need
from the sysimage handle at once.

This allows sysimage loading to be separated into three phases:
  1. Lookup the raw sysimage buffer as a `jl_image_buf_t`
2. For .so sysimages, parse the sysimage and initialize JIT targets,
etc.
  3. Finally load the sysimage into a `jl_image_t`

Care was taken to preserve the existing behavior of calling
`jl_set_sysimg_so` to configure the sysimage before initializing Julia,
although this is likely to be next on the chopping block in a follow-up.

Dependent on #57523.

gbaraldi committed:
Don't use sigatomic before current task is available

aviatesk committed:
improve robustness of `Vararg` checks in newly added `abstract_eval_xxx` (#58325)

Otherwise these subroutines may raise `unhandled Vararg` errors when
compiling e.g. CassetteOverlay for complex call graphs.

aviatesk committed:
use `src.nargs` for `validate_code!` (#58327)

This is a follow-up to https://github.com/JuliaLang/julia/pull/54341.
Otherwise, `validate_code!` may raise wrong errors for unmatched `nargs`
information between generated code and original method.

KristofferC committed:
Apply some of the trimming patches to 1.12 (#58321)

KristofferC commented:
I don't understand how those very old Pkg tests started failing all of a sudden..

KristofferC committed:
Revert "bump Pkg to latest v1.12"

This reverts commit 925d41c7c5852a7b457d892ffb40c53b059e743f.

KristofferC commented:
I reverted Pkg for now so we can get this out earlier since quite a few more packages passes with this.
